searx/engines/google_images.py

```diff
@@ -64,25 +64,29 @@ def request(query, params):
         + google_info['subdomain']
         + '/search'
         + "?"
-        + urlencode(
-            {
-                'q': query,
-                'tbm': "isch",
-                **google_info['params'],
-                'asearch': 'isch',
-                'async': '_fmt:json,p:1,ijn:' + str(params['pageno']),
-            }
-        )
+        + urlencode({'q': query, 'tbm': "isch", **google_info['params'], 'asearch': 'isch'})
+        # don't urlencode this because wildly different AND bad results
+        # pagination uses Zero-based numbering
+        + '&'
+        + 'async=_fmt:json,p:1,ijn:'
+        + str(int(params['pageno'] - 1))
     )
 
     if params['time_range'] in time_range_dict:
         query_url += '&' + urlencode({'tbs': 'qdr:' + time_range_dict[params['time_range']]})
     if params['safesearch']:
         query_url += '&' + urlencode({'safe': filter_mapping[params['safesearch']]})
     params['url'] = query_url
-
     params['cookies'] = google_info['cookies']
     params['headers'].update(google_info['headers'])
+    # this ua will allow getting ~50 results instead of 10. #1641
+    params['headers'].update(
+        {
+            "User-Agent": f"NSTN/3.60.474802233.release Dalvik/2.1.0 (Linux; U; Android 12; \
```

Comment:
It seems like we had already used that user agent earlier but @return42 changed it in https://github.com/searxng/searxng/commit/249989955497cd048fa3312d115971282983b269#diff-7f888885ad7eb66947d95abd3ffa0dabe8a1faa6b1a4b5d4ea0b40908bf0afa6.

@return42 can you still remember why?

Reply:
> @return42 can you still remember why?

nope .. hm, may it was accidentally removed when I forgot to read ..

- https://github.com/searxng/searxng/issues/1641#issue-1333642470